discovery.go

```diff
@@ -29,7 +64,10 @@ func init() {
 
 func main() {
 	log.SetFlags(0)
-	handling.Setup()
+	viper.Set("etcd", mustHostOnlyURL(viper.GetString("etcd")))
+	viper.Set("host", mustHostOnlyURL(viper.GetString("host")))
+
+	handling.Setup(viper.GetString("etcd"))
```

Comment:
That seems reasonable. I amended the commit; now `viper.Get()` is only called from within `discovery.go`.